File: test/elixir/hologram/ex_js_consistency/erlang/elixir_config_test.exs
defmodule Hologram.ExJsConsistency.Erlang.ElixirConfigTest do
  @moduledoc """
  IMPORTANT!
  Each Elixir consistency test has a related JavaScript test in test/javascript/erlang/elixir_config_test.mjs
  Always update both together.
  """

  use Hologram.Test.BasicCase, async: true

  @moduletag :consistency

  describe "identifier_tokenizer/0" do
    test "names the tokenizer Elixir ships with" do
      assert :elixir_config.identifier_tokenizer() == String.Tokenizer
    end
  end
end

File: test/javascript/erlang/elixir_config_test.mjs
"use strict";

import {assert, defineRuntimeGlobals} from "../support/helpers.mjs";

import Erlang_Elixir_Config from "../../../assets/js/erlang/elixir_config.mjs";
import Type from "../../../assets/js/type.mjs";

defineRuntimeGlobals();

// IMPORTANT!
// Each JavaScript test has a related Elixir consistency test in test/elixir/hologram/ex_js_consistency/erlang/elixir_config_test.exs
// Always update both together.

describe("Erlang_Elixir_Config", () => {
  describe("identifier_tokenizer/0", () => {
    const identifier_tokenizer = Erlang_Elixir_Config["identifier_tokenizer/0"];

    it("names the tokenizer Elixir ships with", () => {
      assert.deepStrictEqual(
        identifier_tokenizer(),
        Type.alias("String.Tokenizer"),
      );
    });
  });
});
